projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts

```diff
@@ -1,36 +1,58 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
 import {
-  AfterContentChecked,
-  ChangeDetectionStrategy,
-  Component,
-} from '@angular/core';
-import { Product, TranslationService, WindowRef } from '@spartacus/core';
-import { BehaviorSubject, Observable } from 'rxjs';
+  EventService,
+  Product,
+  TranslationService,
+  WindowRef,
+} from '@spartacus/core';
+import {
+  ComponentCreateEvent,
+  ComponentDestroyEvent,
+} from '@spartacus/storefront';
+import { defer, Observable, of, merge } from 'rxjs';
+import { filter, mapTo } from 'rxjs/operators';
 import { CurrentProductService } from '../current-product.service';
 
 @Component({
   selector: 'cx-product-intro',
   templateUrl: './product-intro.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ProductIntroComponent implements AfterContentChecked {
+export class ProductIntroComponent {
   product$: Observable<Product | null> =
     this.currentProductService.getProduct();
 
-  areReviewsAvailable$ = new BehaviorSubject<boolean>(false);
+  /**
+   * Observable that checks the reviews component availability on the page.
+   */
+  protected areReviewsAvailable$ = merge(
+    // Check if reviews component is already defined:
+    defer(() => of(!!this.getReviewsComponent())),
+
+    // Observe EventService for reviews availability:
+    this.eventService.get(ComponentCreateEvent).pipe(
+      filter((event) => event.id === this.reviewsComponentId),
+      mapTo(true)
+    ),
+    this.eventService.get(ComponentDestroyEvent).pipe(
+      filter((event) => event.id === this.reviewsComponentId),
+      mapTo(false)
+    )
+  );
 
-  protected reviewsTranslationKey =
-    'TabPanelContainer.tabs.ProductReviewsTabComponent';
+  protected reviewsComponentId = 'ProductReviewsTabComponent';
+
+  protected reviewsTranslationKey = `TabPanelContainer.tabs.${this.reviewsComponentId}`;
+
+  protected pageLayoutSelector = 'cx-page-layout.ProductDetailsPageTemplate';
 
   constructor(
     protected currentProductService: CurrentProductService,
     protected translationService: TranslationService,
+    protected eventService: EventService,
     protected winRef: WindowRef
```

Comment:
For automated constructor migrations to work, we have to add new dependencies at the end
```suggestion
    protected winRef: WindowRef,
    protected eventService: EventService
```